tools/openiris_setup.py

```diff
@@ -0,0 +1,816 @@
+                        response_buffer += data
+                        
+                        # Show raw data for debugging
+                        if self.debug and data.strip():
+                            print(f"📡 Raw: {repr(data)}")
+                            print(f"📝 Buffer: {repr(response_buffer[-200:])}")
+                        
+                        # Clean buffer and look for JSON
+                        import re
+                        
+                        # Remove ANSI escape sequences
+                        clean_buffer = re.sub(r'\x1b\[[0-9;]*m', '', response_buffer)
+                        clean_buffer = clean_buffer.replace('\r', '')
+                        
+                        # Look for JSON objects - handle both single-line and multi-line
+                        # Try to find complete JSON objects
+                        start_idx = clean_buffer.find('{')
+                        while start_idx >= 0:
+                            # Count braces to find complete JSON
+                            brace_count = 0
+                            end_idx = -1
+                            
+                            for i in range(start_idx, len(clean_buffer)):
+                                if clean_buffer[i] == '{':
+                                    brace_count += 1
+                                elif clean_buffer[i] == '}':
+                                    brace_count -= 1
+                                    if brace_count == 0:
+                                        end_idx = i + 1
+                                        break
+                            
+                            if end_idx > start_idx:
+                                json_str = clean_buffer[start_idx:end_idx]
+                                
+                                # Try to parse any complete JSON object
+                                try:
+                                    # Clean up the JSON
+                                    clean_json = json_str.replace('\t', ' ').replace('\n', ' ').replace('\r', '')
+                                    clean_json = re.sub(r'\s+', ' ', clean_json)
+                                    
+                                    response = json.loads(clean_json)
+                                    
+                                    # Return if this is a command response with results
+                                    if "results" in response:
+                                        return response
+                                        
+                                except json.JSONDecodeError:
+                                    pass
+                                
+                                # Look for next JSON object
+                                start_idx = clean_buffer.find('{', end_idx)
+                            else:
+                                # No complete JSON found yet
+                                break
```

Comment:
Stuff like that we can move out to a separate method, just to keep things simpler